TextFormatter.java

```diff
@@ -14,13 +17,74 @@ public static void main(String[] args) {
   }
 
   // Konstruktor
-  public TextFormatter(int maxLineLength) {
-    // ...
+  public TextFormatter(int newMaxLineLength) {
+    maxLineLength = newMaxLineLength;
   }
 
   // Ausgabe
   public void print(String aText) {
-    System.out.println("Hier sollte der Text mit passendem Umbruch erscheinen.");
+    String formattedText = formatRightBound(aText);
+
+    System.out.println(formattedText);
+  }
+
+  private String formatLeftBound(String aText) {
+    StringTokenizer st1 =  new StringTokenizer(text, " "); 
+    String formattedText = "";
+    String line = "";
+
+    while (st1.hasMoreTokens()) 
+    { 
+      String nextWord = st1.nextToken();
+      // if the line + nextWord is longer than maxLineLength, create a new line 
+      // for nextWord. The current line is thus complete and added to formattedText.
+      if((line.length() + nextWord.length()) > maxLineLength){
+        formattedText += line + "\n";
+        line = "";
+      }
+      line += nextWord + " "; 
+    }
+     
+    formattedText += line;
+
+    return formattedText;
   }
 
+
+  private String formatRightBound(String aText) {
+    StringTokenizer st1 =  new StringTokenizer(text, " "); 
+    String formattedText = "";
+    String line = "";
+
+    while (st1.hasMoreTokens()) 
+    { 
+      String nextWord = st1.nextToken();
+      // if the line + nextWord is longer than maxLineLength, create a new line 
+      // for nextWord. The current line is thus complete and added to formattedText.
+      if((line.length() + nextWord.length()) > maxLineLength){
+
+        // pads the line with spaces so the line lenght matches maxLineLength
+        if(line.length() < maxLineLength)
+        {
+          //  str.append("point");
+          int amount = maxLineLength - line.length();
+          String padding = "";
+          for(int i = 0; i < amount; i++) 
+          {
+            padding += " ";
+          }
+          line = padding + line;
+        
+        }
+
+        formattedText += line + "\n";
+        line = "";
+      }
+      line += nextWord + " "; 
+    }
+     
+    formattedText += line;
+
+    return formattedText;
+  }
```

Comment:
Die beiden Funktionen formatLeftBound und formatRightBound sind sehr lang zudem noch ähnlich. Ist es vielleicht möglich, die Abweichungen in eigene Methoden zu extrahieren? Entweder durch Unterklassenbildung oder auch steuernde Flags?